src/uct/ib/rc/base/rc_ep.c

```diff
@@ -391,6 +391,22 @@ void uct_rc_txqp_purge_outstanding(uct_rc_txqp_t *txqp, ucs_status_t status,
     }
 }
 
+ucs_status_t uct_rc_ep_flush(uct_rc_ep_t *ep, int16_t max_available)
+{
+    uct_rc_iface_t *iface = ucs_derived_of(ep->super.super.iface, uct_rc_iface_t);
+
+    if (!uct_rc_iface_has_tx_resources(iface) || !uct_rc_ep_has_tx_resources(ep)) {
+        return UCS_ERR_NO_RESOURCE;
+    }
+
+    if (uct_rc_txqp_available(&ep->txqp) == max_available) {
```

Comment:
minor: we could check whether arb group is empty here as well. Then this check can be made first (prior to ep and iface resources) to have shorter path for the case when everything is flushed. However not sure it will be faster wrt to ep->arb_group access.